### PR TITLE
[Core] subprops improvements

### PR DESCRIPTION
**Pull request by philbucher**
this improves the ReadmaterialsUtility

### CODE REVIEW

File: kratos/utilities/read_materials_utility.cpp [hunk 1]
```diff
@@ -296,66 +280,67 @@ void ReadMaterialsUtility::CreateProperty(
...
+        // Copy of the current parameters
+        Parameters sub_prop = SubPropertiesData[i_sub_prop];
+
+        // Define subproperties
+        Properties::Pointer p_new_sub_prop = nullptr;
+        if (sub_prop.Has("use_existing_property")) {
+            // We get the adress if any
+            const std::string& r_use_existing_property = sub_prop["use_existing_property"].GetString();
+
+            // TODO why cannot we directly ask the modelpart with the address?
```
Comment (philbucher) [inline kratos/utilities/read_materials_utility.cpp:304]: @loumalouomega why did you split up the following?
Why cannot you directly ask for the property from the modelpart with the address? That would make it much simpler

**Review by philbucher — COMMENTED** [review]
hiding the whitespaces makes it easier to read

Now the logic is easier to follow

File: kratos/utilities/read_materials_utility.cpp (hunk 1 above)
Comment (loumalouomega) [reply kratos/utilities/read_materials_utility.cpp:304]: Because this part of the code is previous to that :P

**Review by loumalouomega — APPROVED** [review]
OK, if tests run for me OK, certantly I was completely lost after s many patches during the last year

**Comment by philbucher** [comment]
ok then I will merge

**Pull request merged**